src/main/scala/zio/sqs/producer/Producer.scala

```diff
@@ -0,0 +1,196 @@
+package zio.sqs.producer
+
+import java.util.function.BiFunction
+
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import software.amazon.awssdk.services.sqs.model._
+import zio.clock.Clock
+import zio.duration.Duration
+import zio.sqs.serialization.Serializer
+import zio.stream.{ Sink, Stream, ZSink, ZStream }
+import zio._
+
+import scala.jdk.CollectionConverters._
+
+trait Producer[T] {
+
+  def produceE(e: ProducerEvent[T]): Task[ErrorOrEvent[T]]
+
+  def produce(e: ProducerEvent[T]): Task[ProducerEvent[T]]
+
+  def produceBatchE(es: Iterable[ProducerEvent[T]]): Task[List[ErrorOrEvent[T]]]
+
+  def produceBatch(es: Iterable[ProducerEvent[T]]): Task[List[ProducerEvent[T]]]
+
+  def sendStreamE: Stream[Throwable, ProducerEvent[T]] => ZStream[Clock, Throwable, ErrorOrEvent[T]]
+
+  def sendStream: Stream[Throwable, ProducerEvent[T]] => ZStream[Clock, Throwable, ProducerEvent[T]]
+
+  def sendSink: ZSink[Any, Throwable, Nothing, Iterable[ProducerEvent[T]], Unit]
+}
+
+object Producer {
+
+  def make[R, T](
+    client: SqsAsyncClient,
+    queueUrl: String,
+    serializer: Serializer[T],
+    settings: ProducerSettings = ProducerSettings()
+  ): ZManaged[R with Clock, Throwable, Producer[T]] = {
+    val eventQueueSize = nextPower2(settings.batchSize)
+    for {
+      eventQueue <- Queue.bounded[SqsRequestEntry[T]](eventQueueSize).toManaged(_.shutdown)
+      failQueue  <- Queue.bounded[SqsRequestEntry[T]](eventQueueSize).toManaged(_.shutdown)
+      reqRunner  = runSendMessageBatchRequest[R, T](client, failQueue, settings.retryDelay, settings.retryMaxCount) _
```

Comment:
Is using this recommended when we send a single message?

Reply:
depends on the use-case...
the current implementation tries to batch the messages.

if batching are not expected, `batchSize`: could be set to `1` in `ProducerSettings` and behavior will be the same in general.

Reply:
going to add a short explanation to readme.
if there any better way, or there should be an alternative, let me know.